### PR TITLE
Check apps configuration when generating url

### PR DESCRIPTION
**Pull request by Johaney-s**
* When generating url for mail notification about changing preferred email, the domain must be included in the config file
* For new GUI it shouldn't add /krb/gui/ part in the url as that would invalidate it

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java [hunk 1]
```diff
@@ -2318,4 +2328,21 @@ private static String getUserPreferredEmail(PerunSession sess) {
 		}
 	}
 
+	/**
+	 * Returns configuration's brand in which the provided domain is specified
+	 * @param domain
+	 * @return brand or null if domain is not present in the configuration
+	 */
+	private static PerunAppsConfig.Brand getBrandContainingDomain(String domain) {
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java:2336]: This kind of search function could be part of `PerunAppsConfig` itself, what do you think @Vojtech-Sassmann ?

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java [hunk 2]
```diff
@@ -1016,9 +1016,19 @@ public static void sendValidationEmail(User user, String url, String email, int
 		String linkLocation = "/gui/";
 		try {
 			URL urlObject = new URL(url);
+			String domain = urlObject.getProtocol()+"://"+urlObject.getHost();
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java:1019]: This solves the issue for now. But we still rely on `referer` header to be sent. In the end of this dev cycle we want GUI to send us domain and authorization filter to the API, so just keep that in mind. I'm not sure, if it should or shouldn't be part of this pull-request.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java (hunk 1 above)
Comment (Vojtech-Sassmann) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java:2336]: Yeah, definitely.

Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java]: Okay, moved to `PerunAppsConfig`.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java (hunk 2 above)
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java:1021]: So now it should be possible to either pass referer or verification link url, authentication can be passed as query parameter too, the old way of appending it to path is still possible for now, might want to delete later?

**Review by Vojtech-Sassmann — CHANGES_REQUESTED** [review]

**Review by Vojtech-Sassmann — APPROVED** [review]
I think this should be working. We can test this with the changes that have to be done to the GUI applications.

**Pull request merged**